File: web/app/lib/types/feed.ts
export interface FormState {
  title: string;
  description: string;
  privacy: string;
  image: File | null;
  privateUsers: string[];
}

export interface PostFormProps {
  onCancel: () => void;
  onPostCreated?: () => void;
  inGroup: boolean;
  groupId?: string;
}

export interface CreatePostFormProps {
  onPostCreated?: () => void;
}

export interface followingUsers {
  id: string;
  first_name: string;
  last_name: string;
}

export interface PrivateSectionProps {
  users: followingUsers[];
  privateUsers: string[];
  toggleUser: (userId: string) => void;
}

export type PostErrors = {
  title?: string;
  description?: string;
  privacy?: string;
  private?: string;
  duplicate?: string;
  coverImage?: [] | string;
};

export interface PostType {
  id: number;
  author: {
    id: string;
    nickname: string;
    fistname: string;
    lastname: string;
    avatar?: string;
  };
  group: {
    group_id?: number;
    group_title?: string;
  };
  title: string;
  description: string;
  privacy: string;
  image_url?: string;
  created_at: string;
  is_liked: boolean;
  total_likes: number;
  total_comments: number;
}

export interface CommentState {
  content: string;
  postId: number;
  image: File | null;
}

export type CommentErrors = {
  content?: string;
  coverImage?: string;
};

export interface CommentType {
  id: number;
  post_id: number;
  author: {
    id: string;
    nickname: string;
    fistname: string;
    lastname: string;
    avatar?: string;
  };
  content: string;
  image_url?: string;
  created_at: string;
}

export interface PostListProps {
  refreshKey?: number;
  fetchData: (cursor: number) => Promise<any>; 
}

export interface CommentSectionProps {
  postId: number;
  onCommentCreated: () => void;
}

File: internal/types/feed.go
package types

import "time"

type Post struct {
	Id          int       `json:"id"`
	UserId      string    `json:"user_id"`
	GroupId     *int      `json:"group_id"`
	Title       string    `json:"title"`
	Description string    `json:"description"`
	Privacy     string    `json:"privacy"`
	ImageUrl    *string   `json:"image_url"`
	CreatedAt   time.Time `json:"created_at"`
}

type Comment struct {
	Id        int       `json:"id"`
	UserId    string    `json:"user_id"`
	PostId    int       `json:"post_id"`
	Content   string    `json:"content"`
	ImageUrl  *string   `json:"image_url"`
	CreatedAt time.Time `json:"created_at"`
}

type Reaction struct {
	UserId string `json:"user_id"`
	PostId int    `json:"post_id"`
}

type UserResponse struct {
	Id        string `json:"id"`
	Nickname  string `json:"nickname"`
	FirstName string `json:"fistname"`
	LastName  string `json:"lastname"`
	Avatar    string `json:"avatar"`
}

type PostResponse struct {
	Id            int          `json:"id"`
	User          UserResponse `json:"author"`
	GroupId       *int         `json:"group_id"`
	Title         string       `json:"title"`
	Description   string       `json:"description"`
	Privacy       string       `json:"privacy"`
	ImageUrl      *string      `json:"image_url"`
	CreatedAt     time.Time    `json:"created_at"`
	IsLiked       bool         `json:"is_liked"`
	TotalLikes    int          `json:"total_likes"`
	TotalComments int          `json:"total_comments"`
}

type CommentResponse struct {
	Id        int          `json:"id"`
	PostId    int          `json:"post_id"`
	User      UserResponse `json:"author"`
	Content   string       `json:"content"`
	ImageUrl  *string      `json:"image_url"`
	CreatedAt time.Time    `json:"created_at"`
}

type PostInput struct {
	UserId       string
	GroupId      *int
	Title        string
	Description  string
	Privacy      string
	ImageUrl     *string
	PrivateUsers []string
}

type CommentInput struct {
	UserId   string
	PostId   int
	Content  string
	ImageUrl *string
}
